CMakeLists.txt

```diff
@@ -8,7 +8,7 @@
 
 cmake_minimum_required(VERSION 3.16)
 
-project(iDynTree VERSION 3.2.1
+project(iDynTree VERSION 3.3.0
```

Comment:
Can you bump to 4.0.0 ? Adding new virtual methods can be an ABI-breaking change on some compilers.

Reply:
https://github.com/robotology/idyntree/issues/823

Reply:
Good point @GiulioRomualdi , feel free to open a PR if you like, otherwise I will address it before releasing 4.0.0 .